### PR TITLE
Raw reader cru

### PR DESCRIPTION
**Pull request by wiechula**
Add possibility to read events in HB scaling mode

### CODE REVIEW

File: Detectors/TPC/reconstruction/include/TPCReconstruction/RawReaderCRU.h [hunk 1]
```diff
@@ -35,7 +37,158 @@ namespace o2
...
+  /// \struct EventInfo
+  /// \brief helper to store event information
+  struct EventInfo {
+    EventInfo() : CRUInfoArray(CRU::MaxCRU) {}
+    EventInfo(uint32_t heartbeatOrbit) : HeartbeatOrbit{heartbeatOrbit}, CRUInfoArray(CRU::MaxCRU) {}
+    //EventInfo() {}
+    //EventInfo(uint32_t heartbeatOrbit) : HeartbeatOrbit{heartbeatOrbit} {}
+    EventInfo(const EventInfo&) = default;
+
+    bool operator<(const EventInfo& other) { return HeartbeatOrbit < other.HeartbeatOrbit; }
```
Comment (shahor02) [inline Detectors/TPC/reconstruction/include/TPCReconstruction/RawReaderCRU.h]: This method must be ``const``

**Comment by wiechula** [comment]
Hi Ruben, thanks for the comments. They are implemented now.

**Comment by shahor02** [comment]
@sawenzel the checkode complains about the malformed license notice in https://github.com/AliceO2Group/AliceO2/tree/dev/Detectors/TPC/calibration/include/TPCCalibration/CalibRawBase.h which is not touched by this PR and has exactly the same license as other headers (compared with diff). Any idea why?

**Comment by shahor02** [comment]
@wiechula there is also a Travis CI complaint, could you pass the PR by clang-format and push again. o2checkcode has error reports which look like bogus, it is worth trigger a new check.

**Comment by sawenzel** [comment]
I don't know why the header message test fails. Maybe the check was triggered just by this PR and we did not see the problem earlier. But there seems to be another real problem (which I thought we had solved in the past):
```
Error while processing /mnt/mesos/sandbox/sandbox/O2/Steer/DigitizerWorkflow/src/SimpleDigitizerWorkflow.cxx.
/mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/GCC-Toolchain/v7.3.0-alice1-2/bin/../lib/gcc/x86_64-unknown-linux-gnu/7.3.0/../../../../include/c++/7.3.0/variant:238:46: error: cannot cast 'std::variant<std::function<bool (const o2::framework::DataRef &)>, std::function<std::tuple<o2::framework::MakeRootTreeWriterSpec::TerminationCondition::Action, bool> (const o2::framework::DataRef &)> >' to its private base class 'std::__detail::__variant::_Variant_storage<false, std::function<bool (const o2::framework::DataRef &)>, std::function<std::tuple<o2::framework::MakeRootTreeWriterSpec::TerminationCondition::Action, bool> (const o2::framework::DataRef &)> >' [clang-diagnostic-error]
      return __get(std::in_place_index<_Np>, std::forward<_Variant>(__v)._M_u);
```

@matthiasrichter : Does this ring a bell?

**Comment by shahor02** [comment]
@sawenzel I saw this but this MakeRootTreeWriterSpec error was reported only in the checkcode of this PR, so might be some malfunctioning. That's why I asked @wiechula to push after clang-formated to trigger another check.

**Comment by wiechula** [comment]
@sawenzel, @shahor02 I pushed the clang-format. From my side this can be merged

File: Detectors/TPC/calibration/include/TPCCalibration/CalibRawBase.h [hunk 2]
```diff
@@ -1,4 +1,5 @@
 // Copyright CERN and copyright holders of ALICE O2. This software is
+// setEventInfo = true;
```
Comment (shahor02) [inline Detectors/TPC/calibration/include/TPCCalibration/CalibRawBase.h]: @wiechula One of the checkcode failures is due to this stray line (@sawenzel this is clarified, sorry for false alarm).

**Comment by shahor02** [comment]
The checkcode still fails (@matthiasrichter  ?) 
````
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/build_o2checkcode_o2/0/Framework/Utils/include/DPLUtils/MakeRootTreeWriterSpec.h:345:32: note: in instantiation of function template specialization 'std::get<std::function<std::tuple<o2::framework::MakeRootTreeWriterSpec::TerminationCondition::Action, bool> (const o2::framework::DataRef &)>, std::function<bool (const o2::framework::DataRef &)>, std::function<std::tuple<o2::framework::MakeRootTreeWriterSpec::TerminationCondition::Action, bool> (const o2::framework::DataRef &)> >' requested here
            auto& check = std::get<TerminationCondition::CheckProcessing>(terminationCondition.check);
                               ^
````

**Comment by sawenzel** [comment]
Our clang/llvm is probably too old. This should go away when we move to the new version (already in preparation).

**Pull request merged**